application/libraries/Save.php

```diff
@@ -182,26 +119,29 @@ function saveSurvey()
 
         $aSaveForm  = array();
         $timeadjust = getGlobalSetting('timeadjust');
+        $this->saveData = array(
```

Comment:
Does this have to be an object property? What about passing around a DTO instead?

Reply:
DTO = data transfer object

Reply:
Why ? It's clearly an array of value filled by POST, allowed to be updated in `PluginEvent('saveSurveyForm')`

Why it can be used as an object ?

I really think it's not an object. It's an array of value sent by user, updatable by plugin. Moved `$_POST` to `$this->saveData`

Reply:
I don't like when devs use object properties just because it's easy to write. Sending DTOs as argument to functions more clearly communicates what the function needs to do a task. IMO. :) 

Basically never use associative arrays: https://steemit.com/php/@crell/php-use-associative-arrays-basically-never

Reply:
$_POST is an associative array.

Dynamic form by Yii use associative array too (because came from POST value).

Reply:
I don't want to make it complex for plugin dev .
It's here to allow plugin to update : not more.

If needed i come back to App()->request->getPost …

Reply:
Yeah, never use $_POST, always the request object.

Class is less complex than associative array, always, because it has will-defined properties.

Reply:
I can checkout this PR and check more closely, hard to see on github...

Reply:
About $_POST : see the commit : saveData replace $_POST to allow update by plugin.

I can 4 class variables and send it to plugin event (and allow update)
I can create a class and send it to plugin event and allow update

But i really think it's more clear if it's a array of value here .

Reply:
Bah. I'd prefer if launchSaveFormEvent() returned values instead of setting object properties, but fine. I'll merge.

Reply:
You mean (sample)

`$saveSurveyEvent->set('identifier', $identifier);
$saveSurveyEvent->set('email', $email);`

It's what i done at 1st step, but after review : i think a plugin can ADD some input in sabe system and have it in twig view with saveData.myvalue
I didn't update twig file here, but i do it when testing :/

:)

Reply:
PS : don't merge new event if it can be improved please.

Reply:
Was the manual updated with new event data, btw?

Reply:
Currently not, wait for merge ;)

Reply:
> Was the manual updated with new event data, btw?

https://manual.limesurvey.org/SaveSurveyForm